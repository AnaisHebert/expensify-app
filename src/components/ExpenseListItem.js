import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <tr>
        <td><Link to={`/edit/${id}`}>{description}</Link></td>
        <td>{amount}</td>
        <td>{createdAt}</td>
    </tr>
);

export default ExpenseListItem;
