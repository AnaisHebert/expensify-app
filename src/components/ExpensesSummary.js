import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';


export class ExpensesSummary extends React.Component{
    constructor(props){
        super(props);
    };
    render = () => (
        <div>
            <h3>
                Viewing {this.props.expensesCount} 
                {this.props.expensesCount === 1 ? ' expense ' : ' expenses '}
                totalling {numeral(this.props.expensesTotal / 100).format('$0,0.00')}
            </h3>
        </div>
    );
};

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: expenses.length,
        expensesTotal: expensesTotal(expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);