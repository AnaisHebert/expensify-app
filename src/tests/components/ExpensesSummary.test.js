import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseSummary with 0 expenses totalling $0.00', () => {
    const wrapper = shallow(
        <ExpensesSummary 
            expensesCount={0}
            expensesTotal={0}
        />);
    expect(wrapper).toMatchSnapshot();
});

test('should rander ExpenseSummary with 1 expense totalling $1.95', () => {
    const wrapper = shallow(
        <ExpensesSummary 
            expensesCount={1}
            expensesTotal={195}
        />);
    expect(wrapper).toMatchSnapshot();
});
