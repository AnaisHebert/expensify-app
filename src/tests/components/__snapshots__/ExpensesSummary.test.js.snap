// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rander ExpenseSummary with 1 expense totalling $1.95 1`] = `
<div>
  <h3>
    Viewing 
    1
     expense 
    totalling 
    $1.95
  </h3>
</div>
`;

exports[`should render ExpenseSummary with 0 expenses totalling $0.00 1`] = `
<div>
  <h3>
    Viewing 
    0
     expenses 
    totalling 
    $0.00
  </h3>
</div>
`;
